### ISSUE TITLE
Fatal error: Cannot find module 'validator'

### ISSUE THREAD

**Issue by irlnathan** [tilomitra/bedrock#3]
First, great project!

When executing `grunt db:migrate:up` I get `Fatal error: Cannot find module 'validator'`.

**Comment by tilomitra** [comment]
Can you try doing an `npm install validator --save`?

**Comment by irlnathan** [comment]
Executing

`npm install validator --save-dev`

returns:
```
npm WARN package.json Bedrock@2.1.0 No license field.
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "validator" "--save-dev"
npm ERR! node v4.2.3
npm ERR! npm  v2.14.7
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package grunt@1.0.1 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer grunt-contrib-clean@1.0.0 wants grunt@>= 0.4.5
npm ERR! peerinvalid Peer grunt-contrib-coffee@1.0.0 wants grunt@>= 0.4.5
npm ERR! peerinvalid Peer grunt-contrib-concat@1.0.1 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-contrib-sass@1.0.0 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-debug-task@0.1.8 wants grunt@~0.4.2
npm ERR! peerinvalid Peer grunt-react@0.12.3 wants grunt@~0.4.2
npm ERR! peerinvalid Peer grunt-sass@2.0.0 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-bump@0.8.0 wants grunt@>=1.0.1
npm ERR! peerinvalid Peer grunt-eslint@19.0.0 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-concurrent@2.3.1 wants grunt@>=0.4.0
```

**Comment by tilomitra** [comment]
That's quite odd. Can you tell me what environment you are running this in? OSX Version? Windows? etc.

**Comment by irlnathan** [comment]
OS X El Capitan `10.11.6`
Node version: `4.2.3`
npm version: `2.14.7`
grunt-cli: `1.2.0`

**Comment by tilomitra** [comment]
@irlnathan Can you try it with v2.2.0? I just released it. Remove your node modules and re-install everything.

```
rm -rf node_modules
npm i
```
Let me know if that works. I removed some grunt tasks that were messing up.

**Comment by irlnathan** [comment]
I'm now using `npm v2.2.0` and after reinstalling `node_modules` I continue to receive the error `Fatal error: Cannot find module 'validator'.` after executing `grunt db:migrate:up`. Also at the end of the installation I get the same error I did when trying to install the `validator` module:

```
Testing binary
Binary is fine
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.2.3
npm ERR! npm  v2.2.0
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package grunt does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer grunt-contrib-clean@1.0.0 wants grunt@>= 0.4.5
npm ERR! peerinvalid Peer grunt-contrib-coffee@1.0.0 wants grunt@>= 0.4.5
npm ERR! peerinvalid Peer grunt-contrib-concat@1.0.1 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-contrib-sass@1.0.0 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-debug-task@0.1.8 wants grunt@~0.4.2
npm ERR! peerinvalid Peer grunt-react@0.12.3 wants grunt@~0.4.2
npm ERR! peerinvalid Peer grunt-sass@2.0.0 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-bump@0.8.0 wants grunt@>=1.0.1
npm ERR! peerinvalid Peer grunt-concurrent@2.3.1 wants grunt@>=0.4.0
npm ERR! peerinvalid Peer grunt-eslint@19.0.0 wants grunt@>=0.4.0
```

**Comment by tilomitra** [comment]
@irlnathan So I'm using Node v4.2.2 and npm v3.5.2. I also tried Node v5.12.0 and npm v3.8.6. I think the issue is that your npm is < v3.x. I remember that in v3.x, they flattened the `node_modules` directory, and I'm wondering if this is happening because of that.

Also, in the latest release of Bedrock v2.2.0, I removed `grunt-react` and `grunt-debug-task`. If you look at your error message above, you'll notice that those required `grunt@~0.4.2`, whereas everything else wanted grunt@>0.4.0, and I believe you had grunt > v1.0. 

Basically, if you do a clean `npm install`, (maybe run `npm cache clean; npm install`) you should not see `grunt-react` or `grunt-debug-task` showing up in the error message.

The gist of it is:

- Try to get on npm > v3.x
- Make sure you're on the latest Bedrock (v2.2.0 in package.json)
- Run an `npm cache clean; npm install;` because it looks like you are still pulling in modules that you shouldn't be.

**Comment by irlnathan** [comment]
Upgrading to npm `3.5.2` did the trick...thanks.

**Comment by fadhlaouir** [comment]
In your DockerFiles, the `validator ` package is being installed globally, rather than in the current directory. This is seen here in this command: `npm install -g validator`. You can see that `nodemon` is also being installed globally and doesn't appear in the `package.json` list of dependencies either.

If you want to install this package locally, run `npm i -S validator`.

**Comment by abhishek7553** [comment]
change your directory
